src/client/app_configuration_client.rs

```diff
@@ -73,10 +64,32 @@ pub trait AppConfigurationClient {
     /// will always evaluate the same entities to the same values, no updates
     /// will be received from the server
     fn get_property(&self, property_id: &str) -> Result<PropertySnapshot>;
+}
+
+/// AppConfiguration client for browsing, and evaluating features and properties.
+pub trait AppConfigurationClient: ConfigurationProvider {
+    /// Returns a proxied [`Feature`](crate::Feature).
+    ///
+    /// This proxied feature will envaluate entities using the latest information
+    /// available if the client implementation support some kind of live-updates.
+    fn get_feature_proxy<'a>(&'a self, feature_id: &str) -> Result<FeatureProxy<'a>>;
 
     /// Returns a proxied [`Property`](crate::Property).
     ///
     /// This proxied property will envaluate entities using the latest information
     /// available if the client implementation support some kind of live-updates.
     fn get_property_proxy(&self, property_id: &str) -> Result<PropertyProxy>;
 }
+
+impl<T: ConfigurationProvider> AppConfigurationClient for T {
+    fn get_feature_proxy<'a>(&'a self, feature_id: &str) -> Result<FeatureProxy<'a>> {
+        // FIXME: there is and was no validation happening if the feature exists.
+        // Comments and error messages in FeatureProxy suggest that this should happen here.
+        // same applies for properties.
```

Comment:
I think not extremely important for now. feature evaluation will still fail with error in this case then.

Reply:
The comment was already there. I agree it's not important right now and probably this is the right way to do it. Some feature may exist now and not in new configurations and the other way around. So probably the best is just to validate it when it's going to be evaluated.